CMakeLists.txt

```diff
@@ -287,6 +288,8 @@ set_target_properties(rocksdb${ARTIFACT_SUFFIX} PROPERTIES COMPILE_FLAGS "-DROCK
 add_dependencies(rocksdb${ARTIFACT_SUFFIX} GenerateBuildVersion)
 target_link_libraries(rocksdb${ARTIFACT_SUFFIX} ${LIBS})
 
+include(${CMAKE_CURRENT_SOURCE_DIR}/java/jni.inc)
```

Comment:
Ad1) JNI is already enabled in conditional way. I just have condition inside like in third party. Do you really want to move it? It should be same as for other libraries (condition is inside third-party.Inc)